tests/MultiThreadingTest.cpp

```diff
@@ -0,0 +1,67 @@
+#include "gtest/gtest.h"
+#include <thread>
+#include "ui/controller/DefaultController.h"
+#include "ui/command/Command.h"
+
+class MultiThreadingTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        controller_ = std::make_shared<ui::DefaultController>(
+                std::make_shared<model::Model>(
+                        std::make_shared<model::TaskManager>(std::make_shared<model::IdGenerator>())));
+    }
+
+    std::shared_ptr<ui::command::Command> RandomCommand() {
```

Comment:
what exactly are you testing? 

you made thread safe your Task Manager, you're testing UI?